main.tf

```diff
@@ -19,3 +19,55 @@ resource "local_file" "values" {
   directory_permission = "0755"
   filename             = "${path.root}/${var.argocd.path}/${local.app_name}.yaml"
 }
+
+locals {
+  module_name  = basename(abspath(path.module))
+  app_name     = "external-secrets"
+  cluster_name = var.cluster_name
+  aws_region   = var.aws_region == "" ? data.aws_region.current.name : var.aws_region
+
+  aws_assume_role_arn       = module.iam_assumable_role[0].this_iam_role_arn
+  template_helm_values = templatefile("${path.module}/values/values.yaml",
+    {
+      aws_assume_role_arn = local.aws_assume_role_arn
+      poller_interval     = var.poller_interval
+      aws_region          = local.aws_region
+      app_name            = local.app_name
+  })
+  chart_values = var.chart_values == "" ? local.template_helm_values : var.chart_values
+}
+module "iam_assumable_role" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "~> v3.6.0"
+  count                         = 1
```

Comment:
why do you have count =  1 there?
Since there are no "terraform condition"  ```count``` is redundant in this case.